apps/btshell/src/parse.c

```diff
@@ -0,0 +1,611 @@
+/*
```

Comment:
Btw, a lot of this file (parse.c) has been extracted from bletiny into a new package: util/parse.  I wouldn't expect it to be part of this pull request, but at some point we should think about leveraging that package and removing the duplicate code in these apps.